### ISSUE TITLE
Tense Semantic Effects: Learn temporal meaning

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#102]
## Overview

Tense (past/present/future) is currently captured in the AST as a frozen feature, but it carries significant **semantic meaning** that should be learned.

## Current State

**In AST**: Yes (`tempo` field)
```python
"as": {"vortspeco": "verbo", "tempo": "prezenco"},
"is": {"vortspeco": "verbo", "tempo": "pasinteco"},
"os": {"vortspeco": "verbo", "tempo": "futuro"},
```

**Current approach**: Frozen one-hot feature (no learning)

### Temporal Reference
- **Past** (-is): Event completed, historical fact, certainty
- **Present** (-as): Ongoing, current state, habitual, timeless truth
- **Future** (-os): Prediction, intention, varying certainty

### Examples
- "Li mortis" (He died) - completed event, certain
- "Li mortas" (He is dying) - ongoing process
- "Li mortos" (He will die) - prediction, less certain

### Relevance for Q&A
- "Kiam mortis Zamenhof?" needs to match **past-tense** statements
- "Kio okazos?" needs **future-oriented** content
- Present tense often expresses timeless facts

### 1. Detection (Deterministic) - Already Done
```python
ast["tempo"] = "pasinteco"  # Already in parser
```

### 2. Semantic Effect (Learned)
```python
class TenseTransformer(nn.Module):
    def __init__(self, dim):
        self.tense_transforms = nn.ModuleDict({
            'pasinteco': nn.Linear(dim, dim),
            'prezenco': nn.Linear(dim, dim),
            'futuro': nn.Linear(dim, dim),
        })
    
    def forward(self, embedding, tense):
        return self.tense_transforms[tense](embedding)
```

### 3. Training Pairs
```python
# Same content, different tense - should have graded similarity
("Li venas", "Li venis", similarity=0.7)      # Present → Past
("Li venas", "Li venos", similarity=0.6)      # Present → Future
("Li venis", "Li venos", similarity=0.4)      # Past → Future

# Temporal ordering should be preserved in embedding space
```

### 4. Evaluation Metrics
| Metric | Target |
|--------|--------|
| Tense classification accuracy | >95% |
| Temporal ordering preserved | Past < Present < Future similarity |
| Retrieval by tense | Match question tense to answer tense |

## Scope

1. Keep deterministic detection in parser ✓
2. Add learned tense transformation in sentence encoder
3. Create training pairs for tense semantics
4. Add evaluation metrics

## Related Issues

- #101 Meta: Grammatical Features with Semantic Content (parent)
- #84 Participle structure (also temporal)
- #87 Sentence type (also affects meaning)
